Contest_API.md

```diff
@@ -534,6 +529,34 @@ event feed, also note that these should not be considered proper REST
 endpoints, and that the `event-feed` endpoint is a streaming feed in
 NDJSON format.
 
+Note that `api`, `access`, `account`, `state`, `scoreboard`, and `event-feed`
+are singular nouns and indeed contain only a single object.
+
+### Required and optional endpoints
+
+The only required endpoints are metadata: `api` and `access`.
+The only requirements for properties are that collections must have
+an `id` property, and all endpoints that are supported must
+have at least some property, i.e. an endpoint can't be empty.
+[Referential integrity](#referential-integrity) must also be kept
+(for example, if a submission has a team_id, then teams must be supported).
+
+All other endpoints and properties are optional.
+`access` exists so that you can discover which endpoints and properties
+are supported by a given provider.
+
+In practice there are different types of providers that will offer
+similar sets of endpoints. Some examples:
+ - A contest management system will support at least contests and
+   teams, and may support other configuration endpoints.
+ - A Contest Control System (CCS) will support at least submissions,
+   judgements, and dependencies of these, as well as scoreboard and
+   event-feed.
```

Comment:
That's not at all certain.  You can nitpick some unreasonable example where pretty much all of these examples could be theoretically broken, but a CCS could *absolutely* not support event-feed, and maybe even scoreboard.

Reply:
I agree with what you're saying, but we have these in practise and these are only examples. 
I updated it so that scoreboard is 'likely' and event-feed is 'usual'. Is that downplayed enough?

Reply:
Sure.